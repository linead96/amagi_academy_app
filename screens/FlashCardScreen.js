import React from 'react';
import {
  StyleSheet,
  View,
  PanResponder,
} from 'react-native';
import FlipButton from '../components/FlipButton';
import FlashCard from '../components/FlashCard';
import AnswerChoices from '../components/AnswerChoices';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    fontSize: 20,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  flashcard: {
    flex: 6,
  },
  buttons: {
    flex: 2,
  },
});

export default class FlashCardScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      flipped: false,
      flashCardContent: [
        {
          question: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
          answer: 'Lorem ipsum dolor sit amet.',
          color: '',
        },
        {
          question: 'Donec bibendum velit lorem, ac commodo lacus convallis a.',
          answer: 'Donec bibendum velit lorem.',
          color: '',
        },
        {
          question: 'Integer eu malesuada mi, a lobortis massa.',
          answer: 'Integer eu malesuada mi.',
          color: '',
        },
        {
          question: 'Eget duis at tellus at urna condimentum. Viverra nibh cras pulvinar mattis.',
          answer: 'Viverra nibh cras pulvinar mattis.',
          color: '',
        },
        {
          question: 'Neque aliquam vestibulum morbi blandit cursus. Ut enim blandit volutpat maecenas volutpat blandit.',
          answer: 'Ut enim blandit volutpat maecenas volutpat blandit.',
          color: '',
        },
        {
          question: 'Arcu cursus vitae congue mauris rhoncus aenean.',
          answer: 'Arcu cursus vitae.',
          color: '',
        },
        {
          question: 'Condimentum lacinia quis vel eros donec ac. Interdum velit laoreet id donec.',
          answer: 'Interdum velit laoreet id donec.',
          color: '',
        },
        {
          question: 'Erat pellentesque adipiscing commodo elit. Adipiscing vitae proin sagittis nisl rhoncus mattis rhoncus. ',
          answer: 'Erat pellentesque adipiscing commodo elit.',
          color: '',
        },
        {
          question: 'Nisl purus in mollis nunc sed id semper risus in.',
          answer: 'Nisl purus in mollis nunc.',
          color: '',
        },
        {
          question: 'Volutpat diam ut venenatis tellus in metus vulputate eu.',
          answer: 'Volutpat diam ut venenatis.',
          color: '',
        },
        {
          question: 'Volutpat lacus laoreet non curabitur gravida arcu ac.',
          answer: 'Volutpat lacus laoreet non.',
          color: '',
        },
      ],
      current: 0,
    };

    this._panResponder = PanResponder.create(
      {
        onStartShouldSetPanResponder: () => true,
        onPanResponderEnd: (evt, gestureState) => {
          const { dx } = gestureState;
          const { current, flashCardContent, flipped } = this.state;
          if (dx < -100 && current < flashCardContent.length - 1) {
            this.setState(previousState => (
              {
                current: previousState.current + 1,
              }
            ));
            if (flipped === true) {
              this.setState(previousState => (
                {
                  flipped: !previousState.flipped,
                }
              ));
            }
          } else if (dx > 100 && current > 0) {
            this.setState(previousState => (
              {
                current: previousState.current - 1,
              }
            ));
            if (flipped === true) {
              this.setState(previousState => (
                {
                  flipped: !previousState.flipped,
                }
              ));
            }
          }
        },
      },
    );
  }

  componentDidMount() {
    const { flashCardContent } = this.state;
    if (flashCardContent[0].color === '') {
      const updatedFlashCardContent = flashCardContent.map(
        content => ({ ...content, color: this.generateRandomHexCode() }),
      );

      this.setState({
        flashCardContent: updatedFlashCardContent,
      });
    }
  }

  flippedIt = () => {
    this.setState(previousState => (
      {
        flipped: !previousState.flipped,
      }
    ));
  }

  generateRandomHexCode = () => `#${Math.random().toString(16).slice(2, 8)}`;

  render() {
    const { flipped, flashCardContent, current } = this.state;
    return (
      <View style={styles.container}>
        <View
          style={styles.flashcard}
          {...this._panResponder.panHandlers}
        >
          <FlashCard
            flipped={flipped}
            question={flashCardContent[current].question}
            answer={flashCardContent[current].answer}
            bg={flashCardContent[current].color}
          />
        </View>
        <View style={styles.buttons}>
          <FlipButton
            flip={this.flippedIt}
          />
          <AnswerChoices
            flipped={flipped}
          />
        </View>
      </View>
    );
  }
}
