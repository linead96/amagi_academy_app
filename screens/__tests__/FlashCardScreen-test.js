import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import FlashCardScreen from '../FlashCardScreen';

describe('test flashcardscreen functions and interface', () => {
  test('renders flash card initial screen', () => {
    const tree = renderer.create(<FlashCardScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should change "flipped" state', () => {
    const flashCardComponent = renderer.create(<FlashCardScreen />).getInstance();
    const originalFlip = flashCardComponent.state.flipped;
    flashCardComponent.flippedIt();
    expect(flashCardComponent.state.flipped).toEqual(!originalFlip);
  });

  test('return a random hex code', () => {
    const flashCardComponent = renderer.create(<FlashCardScreen />).getInstance();
    const bg = flashCardComponent.generateRandomHexCode();
    expect(bg).toEqual(expect.stringContaining('#'));
  });
});
