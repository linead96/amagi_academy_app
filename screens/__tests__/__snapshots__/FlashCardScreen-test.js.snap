// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test flashcardscreen functions and interface renders flash card initial screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "fontSize": 20,
      "justifyContent": "center",
    }
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "flex": 6,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#314bde",
          "borderRadius": 30,
          "flex": 6,
          "height": "60%",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": "10%",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "margin": "10%",
            "textAlign": "center",
          }
        }
      >
         
        Question:
Lorem ipsum dolor sit amet, consectetur adipiscing elit.
         
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingLeft": "20%",
          "paddingRight": "20%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
